contracts/InterchainTokenFactory.sol

```diff
@@ -104,7 +104,7 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
     /**
      * @notice Deploys a new interchain token with specified parameters.
      * @dev Creates a new token and optionally mints an initial amount to a specified minter.
-     * The function is marked as `payable` to enable multi-call execution.
+     * The function is `payable` because non-payable functions cannot be called in a multi-call context.
```

Comment:
```suggestion
     * This function is `payable` because non-payable functions cannot be called in a multicall that calls other `payable` functions.
```
Still not quite right. update the rest